Bugfix/dockerfile

cramirezhe opened the pull request:
This fix allows to compile GroundingDINO while building the Docker Image, otherwise we will have warning messages such as https://github.com/IDEA-Research/Grounded-Segment-Anything/issues/53 and  https://github.com/IDEA-Research/Grounded-Segment-Anything/issues/84 and we won't be able to use CUDA inside Docker.

rentainhe commented:
Thank you so much for fixing this issue!

cramirezhe commented:
You’re welcome. 

Maybe it is a good idea to push your Docker image to a public registry.

rentainhe commented:
> You’re welcome.
> 
> Maybe it is a good idea to push your Docker image to a public registry.

Sry about this, there're many users meeting the issue about:

```bash
AttributeError: 'NoneType' object has no attribute 'casefold'
```

Do you have any ideas to fix this? I will remove the modification now, and if you have some more ideas about this you can update a new PR about this issue

rentainhe commented:
I fix this issue in the normal env by setting:

```bash
export AM_I_DOCKER=False
export BUILD_WITH_CUDA=True
```

I think maybe we should add this to README

cramirezhe commented:
I added a default value if the variable is not set:

`am_i_docker = os.environ.get('AM_I_DOCKER', '').casefold() in ['true', '1', 't']`

Also I updated the Makefile to check if NVCC is installed before checking the CUDA version, otherwise the build will fail if NVCC is not installed.
